src/TaskRunnerOptions.ts

```diff
@@ -17,6 +17,21 @@ interface Options {
      */
     onTaskEnd?: (taskName: string) => void;
 
+    /**
+     * An optional callback which will be fired after a task has failed.
+     *
+     * @param {string} taskName
+     */
+    onTaskFail?: (taskName: string) => void;
```

Comment:
Nice, both this and onTaskCancel is handy -- I'm thinking it might be useful to include the error here though, that way you can log things more easily. I'll add that before revving the version and publishing.